### PR TITLE
Feature/virtualized states on ledger

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-engine/src/main/java/com/radixdlt/atom/SubstateTypeId.java [hunk 1]
```diff
@@ -19,23 +19,24 @@
...
+	STAKE_OWNERSHIP((byte) 0x8),
+	PREPARED_UNSTAKE((byte) 0x9),
+	EXITTING_STAKE((byte) 0xa),
+	VALIDATOR_META_DATA((byte) 0xb),
+	VALIDATOR_STAKE_DATA((byte) 0xc),
+	VALIDATOR_BFT_DATA((byte) 0xd),
+	VALIDATOR_ALLOW_DELEGATION_FLAG((byte) 0xe),
+	VALIDATOR_REGISTERED_FLAG_COPY((byte) 0xf),
+	VALIDATOR_RAKE_COPY((byte) 0x10),
+	VALIDATOR_OWNER_COPY((byte) 0x11);
```
Comment (siy) [inline radixdlt-engine/src/main/java/com/radixdlt/atom/SubstateTypeId.java:39]: More changes for ledger app...  :)

File: radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/REInstruction.java [hunk 2]
```diff
@@ -144,7 +182,7 @@ Object read(REParser.ParserState parserState, ByteBuffer buf, SubstateDeserializ
 				return SubstateIndex.create(array, d.byteToClass(array[0]));
 			}
 		},
-		DOWNINDEX((byte) 0xd, REOp.DOWNINDEX) {
+		DOWNINDEX((byte) 0xf, REOp.DOWNINDEX) {
```
Comment (siy) [inline radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/REInstruction.java:185]: Even more changes for ledger app :)

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/SystemConstraintScrypt.java [hunk 3]
```diff
@@ -48,6 +55,28 @@ public SystemConstraintScrypt(Set<String> systemNames) {
...
+	}
+
+	private static class AllocatingVirtualState implements ReducerState {
+		private final LinkedList<SubstateTypeId> substatesToVirtualize = new LinkedList<>();
+		AllocatingVirtualState() {
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_STAKE_DATA);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_ALLOW_DELEGATION_FLAG);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_REGISTERED_FLAG_COPY);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_RAKE_COPY);
+			substatesToVirtualize.add(SubstateTypeId.VALIDATOR_OWNER_COPY);
```
Comment (iamyulong) [inline radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/SystemConstraintScrypt.java:68]: Are we missing `UNCLAIMED_READDRESS`?

**Review by LukasGasior1 — APPROVED** [review]

File: radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/SystemConstraintScrypt.java (hunk 3 above)
Comment (talekhinezh) [reply radixdlt-engine/src/main/java/com/radixdlt/application/system/scrypt/SystemConstraintScrypt.java:68]: No, this is handled in a separate procedure

**Pull request merged**